### PR TITLE
Restructured how connectivity tests are configured and run [RIAK-1608].

### PR DESCRIPTION
**Pull request by tburghart**
Due to leftover configuration attributes from prior tests within this module (see RIAK-1607 (#747)), the order of tests has to be managed.  In order to accomplish this and be able to see exactly what configurations are used for each test, configuration blocks have been more directly associated with each connectivity test.

### CODE REVIEW

File: tests/replication2_ssl.erl [hunk 1]
```diff
@@ -14,221 +49,153 @@ confirm() ->
...
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+            ] ++ CIsite2#ci.ssl }],
+            true},
+        {"SSL connectivity with intermediate CA",
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+            ] ++ CIsite1#ci.ssl }],
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+            ] ++ CIsite3#ci.ssl }],
+            true},
+        {"wildcard certifictes on both ends",
```
Comment (macintux) [inline tests/replication2_ssl.erl]: "certificates"

File: tests/replication2_ssl.erl [hunk 2]
```diff
@@ -14,221 +49,153 @@ confirm() ->
...
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+            ] ++ CIsiteWC#ci.ssl }],
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+            ] ++ CIsiteWC#ci.ssl }],
+            true},
+        {"wildcard certifictes on one end",
```
Comment (macintux) [inline tests/replication2_ssl.erl]: "certificates"

File: tests/replication2_ssl.erl [hunk 3]
```diff
@@ -14,221 +49,153 @@ confirm() ->
...
+        {"disallowing intermediate CA connection",
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+                , {ssl_depth, 0}
+            ] ++ CIsite3#ci.ssl }],
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+                , {ssl_depth, CIsite3#ci.rd}
+            ] ++ CIsite1#ci.ssl }],
+            false},
+        %% first use of peer_common_name_acl, all subsequent tests must specify it
+        {"wildcard certifictes on one end with ACL",
```
Comment (macintux) [inline tests/replication2_ssl.erl]: "certificates"

File: tests/replication2_ssl.erl [hunk 4]
```diff
@@ -14,221 +49,153 @@ confirm() ->
...
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+                , {ssl_depth, CIsite2#ci.rd}
+                , {peer_common_name_acl, [?DOM_WC]}
+            ] ++ CIsite1#ci.ssl }],
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+                , {ssl_depth, CIsite1#ci.rd}
+                , {peer_common_name_acl, [CIsite1#ci.cn]}
+            ] ++ CIsite2#ci.ssl }],
+            true},
+        {"wildcard certifictes on both ends with ACLs",
```
Comment (macintux) [inline tests/replication2_ssl.erl]: "certificates"

File: tests/replication2_ssl.erl [hunk 5]
```diff
@@ -14,221 +49,153 @@ confirm() ->
...
+            ] ++ CIsiteWC#ci.ssl }],
+            true},
+        {"wildcard certifictes on one end",
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+            ] ++ CIsiteWC#ci.ssl }],
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+            ] ++ CIsite3#ci.ssl }],
+            true},
+        %% first use of ssl_depth, all subsequent tests must specify it
+        {"disallowing intermediate CA setting",
```
Comment (macintux) [inline tests/replication2_ssl.erl]: Please tweak the description on this or the next test, confusing two see two nearly identical tests with identical descriptions but one is intended to fail and one passes.

File: tests/replication2_ssl.erl [hunk 6]
```diff
@@ -14,221 +49,153 @@ confirm() ->
...
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+                , {ssl_depth, CIsiteWC#ci.rd}
+                , {peer_common_name_acl, [CIsiteWC#ci.wc]}
+            ] ++ CIsiteWC#ci.ssl }],
+            [ConfRepl, {riak_core, [{ssl_enabled, true}
+                , {ssl_depth, CIsiteWC#ci.rd}
+                , {peer_common_name_acl, [CIsiteWC#ci.wc]}
+            ] ++ CIsiteWC#ci.ssl }],
+            true},
+        {"expired certificates fail",
```
Comment (macintux) [inline tests/replication2_ssl.erl]: Perhaps test expired certificates without an ACL as well?

**Comment by macintux** [comment]
Mostly superficial tests. Passes locally, looks very good.

**Comment by tburghart** [comment]
I'll make those tweaks, and do some re-ordering, and may add a couple of ACL-related ones.

Using a heavily-instrumented riak_core_ssl_util, we're not getting coverage of some of the verification clauses.

**Pull request merged**